stdlib/pervasives.ml

```diff
@@ -420,6 +455,28 @@ let really_input_string ic len =
   really_input ic s 0 len;
   bytes_unsafe_to_string s
 
+let string_of_in_channel ic =
+  let buf = ref (bytes_create 1024) in
+  let len = ref 0 in
+  try
+    while true do
+      (* resize, by doubling the size of [buf] *)
+      if !len = bytes_length !buf then (
+        let new_buf = bytes_create (2 * !len) in
```

Comment:
Some explicit check against max_string_length would be good (better error message).

Also, a tail-rec loop would be better.

(And FWIW, raise_notrace would be better than raise here.)